pyvista/core/objects.py

```diff
@@ -346,11 +346,11 @@ def get_data_range(self, arr=None, preference='row'):
         """
         if arr is None:
             # use the first array in the row data
-            self.GetRowData().GetArrayName(0)
+            arr = self.GetRowData().GetArrayName(0)
         if isinstance(arr, str):
             arr = get_array(self, arr, preference=preference)
         # If array has no tuples return a NaN range
-        if arr is None or arr.size == 0 or not np.issubdtype(arr.dtype, np.number):
+        if arr.size == 0 or not np.issubdtype(arr.dtype, np.number):
```

Comment:
I noticed that this line is correct.
```suggestion
        if arr is None or arr.size == 0 or not np.issubdtype(arr.dtype, np.number):
```